staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -1265,6 +1268,17 @@ func (c *Cacher) getCommonResourceVersionLockedFunc(ctx context.Context, parsedW
 	}
 }
 
+// waitUntilWatchCacheFreshForInitialEvents waits until cache is at least
+// as fresh as given requestedWatchRV and sendInitialEvents was requested.
+func (c *Cacher) waitUntilWatchCacheFreshForInitialEvents(ctx context.Context, requestedWatchRV uint64, opts storage.ListOptions) error {
+	if opts.SendInitialEvents != nil && *opts.SendInitialEvents == true && requestedWatchRV > 0 {
```

Comment:
@wojtek-t I think this case was missing, right?

Note that this method gets the requested RV not the one that is calculated by the cacher.

Reply:
Technically, we probably don't need `requestedWatchRV >0`, because waiting for it is trivial - it's always satisfied.

Reply:
shouldn't we make sure that the cache has seen at least `requestedWatchRV > 0` ?